src/main/java/cms/gov/madie/measure/repositories/MeasureSearchServiceImpl.java

```diff
@@ -38,12 +41,76 @@ private LookupOperation getLookupOperation() {
         .as("measureSet");
   }
 
-  // First get All Active measures
-  // if searchField string is given then search for the searchField string in measureName or eCQM
-  // title
-  // If model is provided filter out those measures based on Model
-  // if draft status is provided then filter out them based on draft value
-  // if filterByCurrentUser = true then filter measures owned by user or shared with
+  private void appendAdditionalSearchCriteriaOmittingcmsId(
+      Criteria measureCriteria, MeasureSearchCriteria measureSearchCriteria) {
+    // Ensure optionalSearchProperties exists and isn’t empty
+    if (CollectionUtils.isNotEmpty(measureSearchCriteria.getOptionalSearchProperties())) {
+      // Build the orOperator for the remaining properties
+      List<Criteria> orConditions = new ArrayList<>();
+      for (String property : measureSearchCriteria.getOptionalSearchProperties()) {
+        // this needs to run whenever we have multiple, however we need to force a search even if
+        // the searchField split is less than 3 if the version is the only category that is applied
+        if (property.equals("version")) {
+          String[] versionParts = measureSearchCriteria.getSearchField().split("\\.");
```

Comment:
You might be able to get rid of the complexity a bit by doing a replace instead and then split where you actually need it.
```
String versionWithoutPeriod = measureSearchCriteria.getSearchField().replace(".", "");
if(versionWithoutPeriod.length == 3) 
if (isNumeric(versionWithoutPeriod)
```